### PR TITLE
vendor: golang.org/x/text v0.3.3 (CVE-2020-14040)

### PR DESCRIPTION
**Pull request by thaJeztah**
full diff: https://github.com/golang/text/compare/v0.3.2...v0.3.3

includes a fix for [CVE-2020-14040](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-14040) (https://github.com/golang/text/commit/23ae387dee1f90d29a23c0e87ee0b46038fbed0e)

### CODE REVIEW

File: vendor/golang.org/x/text/transform/transform.go [hunk 1]
```diff
@@ -648,7 +648,8 @@ func String(t Transformer, s string) (result string, n int, err error) {
 	// Transform the remaining input, growing dst and src buffers as necessary.
 	for {
 		n := copy(src, s[pSrc:])
-		nDst, nSrc, err := t.Transform(dst[pDst:], src[:n], pSrc+n == len(s))
+		atEOF := pSrc+n == len(s)
```
Comment (thaJeztah) [inline vendor/golang.org/x/text/transform/transform.go:651]: I think changes in this file are the only ones related to the CVE

**Comment by thaJeztah** [comment]
@justincormack ptal

**Comment by tiborvass** [comment]
AFAICT (checked all binaries in bundles in some master commit) we are not impacted by this, I could find neither a transform.String nor a utf16decoder string in any of the binaries. For master I don't mind taking this PR though, can be future proof.

**Review by tiborvass — APPROVED** [review]

**Comment by tiborvass** [comment]
in a make shell using 19.03 cli:
```
$ for b in docker dockerd docker-init docker-proxy containerd containerd-shim ctr rootlesskit rootlesskit-docker-proxy runc vpnkit.x86_64; do strings $(which "$b") | grep -E 'transform.String|utf16Decoder'; done
$ echo $?
1
```

**Review by AkihiroSuda — APPROVED** [review]

**Comment by thaJeztah** [comment]
@akihirosuda good to go?

**Pull request merged**